examples/Makefile

```diff
@@ -17,7 +19,7 @@ test_srcs := io_uring-test.c io_uring-cp.c link-cp.c
 test_objs := $(patsubst %.c,%.ol,$(test_srcs))
 
 %: %.c
-	$(QUIET_CC)$(CC) $(CFLAGS) -o $@ $< -luring $(XCFLAGS)
+	$(QUIET_CC)$(CC) $(CPPFLAGS) $(CFLAGS) -o $@ $< -luring $(XCFLAGS)
```

Comment:
I don't find it a good practice passing to CC anything CPP related (i.e. CPPFLAGS).
Better to decouple them

Reply:
Well CC in this case is performing the tasks of the preprocessor, the compiler and the linker, so I don't see why it would not make sense to honor the flags. Notice the options moved to the CPPFLAGS are really preprocessor related options.
In the end CC here is acting as a frontend driving the various phases.

Also notice the default make rules which do honor these various flags:
```shell
make -p -f /dev/null | grep CPPFLAGS
```

Just realized too that for linking invocations we are not passing LDFLAGS,

Reply:
Or from another point of view. I think it is a common expectation that a user can set these flags when building a project and they will affect any invocation involving their build phase. So if as a user I specify say any of CPPFLAGS or CFLAGS or LDFLAGS, then any invocation that would end up calling the relevant backends should end up being affected by them, regardless of these being called as CC, CPP, LD, etc.

Reply:
Ohh blood, I was looking and CXXFLAGS and have got confused in the end.
CPPFLAGS there look sensible, as well as does the whole patch.
LGTM, thanks!